File: 2394-count-subarrays-with-score-less-than-k/2394-count-subarrays-with-score-less-than-k.go
func countSubarrays(nums []int, k int64) int64 {
    n, l := len(nums), 0
    var res, cnt, sum int64
    res = 0
    cnt = 0
    sum = 0
    for r := 0; r < n; r++ {
        sum += int64(nums[r])
        cnt++
        for sum * cnt >= k {
            sum -= int64(nums[l])
            l++
            cnt--
        }
        res += cnt;
    }
    return res
}

File: 2394-count-subarrays-with-score-less-than-k/2394-count-subarrays-with-score-less-than-k.cpp
typedef long long ll;

class Solution {
public:
    long long countSubarrays(vector<int>& nums, long long k) {
        ll res = 0;
        ll sum = 0;
        ll cnt = 0;
        int l = 0, n = nums.size();
        for (int i = 0; i < n; i++) {
            sum += nums[i];
            cnt++;
            while (sum * cnt >= k) {
                sum -= nums[l++];
                cnt--;
            }
            res += cnt;
        }
        return res;
    }
};